Remotion/Web/ClientScript/Scripts/SmartPage.ts

```diff
@@ -464,13 +470,15 @@ class SmartPage_Context
   // Determines whether the page was loaded from cache.
   public CheckIfCached(): void
   {
-    var field = this._theForm.SmartPage_CacheDetectionField;
-    if (field.value == this._cacheStateHasSubmitted)
+    const pageToken = document.getElementById('SmartPage_InstanceToken')!.getAttribute('value');
+    const hasDifferentPageToken = history.state?.pageToken && history.state.pageToken !== pageToken;
+
+    if (history.state?.cacheDetection === this._cacheStateHasSubmitted)
```

Comment:
Since I just spent some time re-familiarzing myself with the sematnics of `_cacheStateHasSubmitted` and `_cacheStateHasLoaded` (before figuring out those are just string constants): should this be updated to use a typed approach with string values defined via discrimininated union / enum? Don't know about the readability impact of this, I'm open for suggestions :)